build.gradle

```diff
@@ -883,6 +884,36 @@ gradle.taskGraph.whenReady({ graph ->
   tasks.getByName('allReports').reportOn rootProject.subprojects.collect{ it.tasks.withType(Test) }.flatten()
 })
 
+task installPulse {
+  def pulseWarName = "pulse-${pulseVersion}.war"
+  def pulseDir = System.env.pulsedir
+
+  if (pulseDir == null || pulseDir.length() == 0) {
+    pulseDir = "${projectDir}/../pulse"
+  }
+
+  if (pulseDir == null || pulseDir.length() == 0) {
+    throw  new Exception(" Pulse directroy can not be empty while building product")
```

Comment:
This check will always fail. Better to check write permission to pulseDir here: `if !(file(pulseDir).canWrite()) { ... }`